lib/iris/plot.py

```diff
@@ -481,6 +482,79 @@ def _get_plot_objects(args):
     return u_object, v_object, u, v, args
 
 
+def _get_geodesic_params(globe):
+    # Derive the semimajor axis and flattening values for a given globe from
+    # its attributes. If the values are under specified, raise a ValueError
+    flattening = globe.flattening
+    semimajor = globe.semimajor_axis
+    try:
+        if semimajor is None:
+            # Has semiminor or raises error
+            if flattening is None:
+                # Has inverse flattening or raises error
+                flattening = 1. / globe.inverse_flattening
+            semimajor = globe.semiminor_axis / (1. - flattening)
+        elif flattening is None:
+            if globe.semiminor_axis is not None:
+                flattening = ((semimajor - globe.semiminor_axis) /
+                              float(semimajor))
+            else:
+                # Has inverse flattening or raises error
+                flattening = 1. / globe.inverse_flattening
+    except TypeError:
+        # One of the required attributes was None
+        raise ValueError('The globe was underspecified.')
+
+    return semimajor, flattening
+
+
+def _shift_plot_sections(u_object, u, v):
```

Comment:
I think it is probably doing something slightly different, but I've added a comment with an alternative algorithm that I would consider working as well for the unrotated PC case.

Comment ref: https://github.com/SciTools/cartopy/issues/368#issuecomment-351409070